thinc/util.py

```diff
@@ -203,12 +203,26 @@ def copy_array(dst: ArrayXd, src: ArrayXd) -> None:  # pragma: no cover
         numpy.copyto(dst, src)  # type: ignore
 
 
-def to_categorical(Y: IntsXd, n_classes: Optional[int] = None) -> FloatsXd:
+def to_categorical(
+    Y: IntsXd,
+    n_classes: Optional[int] = None,
+    *,
+    label_smoothing: float = 0.0,
+) -> FloatsXd:
+    if not 0.0 <= label_smoothing < 0.5:
+        raise ValueError(
+            "label_smoothing should be greater or "
+            "equal to 0.0 and less than 0.5, "
+            "but {} was given".format(label_smoothing)
+        )
     xp = get_array_module(Y)
     if n_classes is None:
         n_classes = int(numpy.max(Y) + 1)  # type: ignore
-    # Unfortunately, cupy does not support put_along_axis.
-    return xp.eye(n_classes, dtype="float32")[Y]
+    label_distr = xp.full(
+        (n_classes, n_classes), label_smoothing / (n_classes - 1), dtype="float32"
+    )
```

Comment:
Let's add a sanity check for `n_classes` here as well, basically that it's bigger than 1, otherwise raise an error?